homeassistant/components/fmi/__init__.py

```diff
@@ -0,0 +1,369 @@
+                        vol.Optional(CONF_OFFSET, default=0): vol.In(FORECAST_OFFSET),
+                        vol.Optional(CONF_MIN_HUMIDITY, default=30): vol.In(
+                            HUMIDITY_RANGE
+                        ),
+                        vol.Optional(CONF_MAX_HUMIDITY, default=70): vol.In(
+                            HUMIDITY_RANGE
+                        ),
+                        vol.Optional(CONF_MIN_TEMP, default=10): vol.In(TEMP_RANGE),
+                        vol.Optional(CONF_MAX_TEMP, default=30): vol.In(TEMP_RANGE),
+                        vol.Optional(CONF_MIN_WIND_SPEED, default=0): vol.In(
+                            WIND_SPEED
+                        ),
+                        vol.Optional(CONF_MAX_WIND_SPEED, default=25): vol.In(
+                            WIND_SPEED
+                        ),
+                        vol.Optional(
+                            CONF_MIN_PRECIPITATION, default=0.0
+                        ): cv.small_float,
+                        vol.Optional(
+                            CONF_MAX_PRECIPITATION, default=0.2
+                        ): cv.small_float,
+                    }
+                )
+            ],
+        )
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Set up the FMI Integration."""
+    hass.data[DOMAIN] = {}
+
+    def setup_fmi_entity(config_map, index=0):
+        try:
+            latitude = config_map[CONF_LATITUDE]
+            longitude = config_map[CONF_LONGITUDE]
+        except KeyError:
+            if hass.config.latitude is None or hass.config.longitude is None:
+                _LOGGER.error(
+                    "Latitude & longitude not set in Home Assistant config and not provided in configuration as well!!"
+                )
+                return False
+            _LOGGER.info(
+                "Using HOME Latitude/Longitude configured in HA for fmi sensor index - %d!",
+                index,
+            )
+            latitude = hass.config.latitude
+            longitude = hass.config.longitude
+
+        try:
+            name = config_map[CONF_NAME]
+            time_step = config_map[CONF_OFFSET]
+            min_temperature = float(config_map[CONF_MIN_TEMP])
+            max_temperature = float(config_map[CONF_MAX_TEMP])
+            min_humidity = float(config_map[CONF_MIN_HUMIDITY])
+            max_humidity = float(config_map[CONF_MAX_HUMIDITY])
+            min_wind_speed = float(config_map[CONF_MIN_WIND_SPEED])
+            max_wind_speed = float(config_map[CONF_MAX_WIND_SPEED])
+            min_precip = float(config_map[CONF_MIN_PRECIPITATION])
+            max_precip = float(config_map[CONF_MAX_PRECIPITATION])
+        except ValueError as v_e:
+            _LOGGER.error("Parameter configuration mismatch - %s", v_e)
+            return False
+        except KeyError as k_e:
+            _LOGGER.error("Parameter key not found - %s", k_e)
+            return False
```

Comment:
Are these errors reachable given your schema above?

Reply:
Good catch! It will not hit this part. I have removed it now.